File: 08/Java/TripleExpression/Abs.java
package TripleExpression;

public class Abs<T> implements TripleExpression<T> {
	private TripleExpression<T> value;
    private Operator<T> op;
    public Abs(TripleExpression<T> val, Operator<T> op) {
        value = val;
        this.op = op;
    }

    public T evaluate(T x, T y, T z) throws Exception {
        T a = value.evaluate(x, y, z);
        return op.abs(a);
    }
}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/Abs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Abs<T> : ITripleExpression<T>
    {
        private ITripleExpression<T> value;
        private IOperator<T> op; 
        public Abs(ITripleExpression<T> value, IOperator<T> op)
        {
            this.value = value;
            this.op = op;
        }

        public T Evaluate(T x, T y, T z)
        {
            T a = value.Evaluate(x, y, z);
            return op.Abs(a);
        }
    }
}
